Fix solutions

dubfib opened the pull request:
You can get the solution from <https://www.nytimes.com/svc/wordle/v2/YYYY-DD-MM.json> and this update just does a get request for today, tomorrow and yesterday using [undici](https://npmjs.com/package/undici) and [moment](https://npmjs.com/package/moment) instead of using a hard coded array. The solutions are also going to be updated in the UTC timezone, not EST and I also changed the workflow from Node v14 to Node v16.

dubfib committed:
Add files via upload

dubfib committed:
make workflow node 16

Commit:
Updated README

dubfib committed:
Merge branch 'main' into main

Commit:
Updated README

Commit:
Updated README

Commit:
Updated README

Commit:
Updated README

audreee commented:
Hi @dubfib! I appreciate the interest in this project. I haven't been actively maintaining this project recently. Thanks for pointing out that the solution can now be found through that url. I went ahead and updated the project to use the new solutions. In general, I personally prefer to minimize dependence on external libraries. Instead, I opted to generate dates using Javascript's native Date class and fetch results with the native node-fetch option.

dubfib commented:
> Hi @dubfib! I appreciate the interest in this project. I haven't been actively maintaining this project recently. Thanks for pointing out that the solution can now be found through that url. I went ahead and updated the project to use the new solutions. In general, I personally prefer to minimize dependence on external libraries. Instead, I opted to generate dates using Javascript's native Date class and fetch results with the native node-fetch option.

Thanks for using my idea, I just refound this repository and attempted to find a way to get the solutions. I also agree on minimizing dependencies and didn't know why I used [moment](https://npmjs.com/package/moment) when I can just use the [Date](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date) class and just use `.getFullYear()`, `.getMonth()` and `.getDay()` and split it between `-` with checks to add a `0` if under day `10`.
```js
function format(date) {
  if (!date) throw new TypeError('Date argument must be non-empty date class');
  
  let year = date.getFullYear();
  let month = date.getMonth() + 1;
  let day = date.getDay();
  
  if (month < 10) month = '0' + month;
  if (day < 10) day = '0' + day;
  
  return year + '-' + month + '-' + day;
};

const date = new Date();
const today = format(date);

date.setDate(date.getDate() + 1);
const tomorrow = format(date);

date.setDate(date.getDate() - 2);
const yesterday = format(date);

console.log(tomorrow, today, yesterday);
//> "2023-02-03" "2023-02-02" "2023-02-01"
```
Also I saw your code and you are using [node-fetch](https://npmjs.com/package/node-fetch), which is fine but Node.js team already made their own fetch library and its called [undici](https://npmjs.com/package/undici) and it's even used in their native fetch library in the newer node versions, you can see the github repository [here](https://github.com/nodejs/undici), and also I'm also pretty sure that [undici](https://npmjs.com/package/undici) is faster than [node-fetch](https://npmjs.com/package/node-fetch) and their syntax is pretty much the same. One thing to note is that [undici](https://npmjs.com/package/undici) is built into node in their newer versions, but if you don't have the newer versions of node you can use the [https](https://nodejs.org/api/https.html) or [http](https://nodejs.org/api/http.html) module instead.
```js
async function fetch(url) {
    if (typeof url !== 'string' || !url) throw new TypeError('URL must be non-empty string!');

    return new Promise(async (resolve, reject) => {
        require('https').get(url, (response) => {
            response.on('data', (data) => resolve(JSON.parse(data.toString())));
            response.on('error', (error) => reject(error));
            response.on('end', () => response.destroy());
        });
    });
};
```
Also you should update your github action from Node.js `v14` to `v18`or `v19` as it `v14` will stop being supported in April of 2023, you can see the release schedule [here](https://github.com/nodejs/release) by the Node.js team themselves. I also figured out one thing about using the endpoint `https://www.nytimes.com/svc/wordle/v2/YYYY-MM-DD.json`, and that it only gives the solution for the current month and the previous ones. That's all I have to say, I write too much..
